src/browser-action-assets/lotterypopup.js

```diff
@@ -0,0 +1,18 @@
+(async function() {
+    // Load the redeem code from local storage
+    const codeStorageKey = "redeemCode";
+    const codeResults = await browser.storage.local.get(codeStorageKey);
+    if(codeResults[codeStorageKey]) {
+        const redeemCode = codeResults[codeStorageKey]
+        const contentElement = document.getElementById("content");
+        contentElement.innerHTML = `<h3><center>Beyond the Paywall</center></h3><p>Congratulations, you've won this week's lottery for participants in the Beyond the Paywall study!<br><br>To claim your $50 Amazon.com gift card, use the code <b>${redeemCode}</b><br><br>Please redeem your code at <a href="https://www.amazon.com/gc/redeem">amazon.com/redeem</a> and then click the "Redeemed" button below to clear this message.</p>`
```

Comment:
The AMO linter is pretty strict about `innerHTML` usage, we might have to change this to be in the HTML file instead. All you have to replace is the `redeemCode` though so it might actually be nicer that way, you could do [.textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) instead.

Reply:
Done